src/js/pages/jobs/JobDetailPage.js

```diff
@@ -0,0 +1,221 @@
+        id: 'more'
+      },
+      {
+        html: 'Suspend',
+        id: 'suspend'
+      },
+      {
+        html: 'Destroy',
+        id: 'destroy'
+      }
+    ];
+
+    return [
+      <button
+        className="button button-inverse button-stroke"
+        key="edit"
+        onClick={this.handleEditButtonClick.bind(this, jobID)}>
+        Edit
+      </button>,
+      <button
+        className="button button-inverse button-stroke"
+        key="run-now"
+        onClick={this.handleRunNowButtonClick.bind(this, jobID)}>
+        Run Now
+      </button>,
+      <Dropdown
+        buttonClassName="dropdown-toggle button button-inverse button-stroke"
+        dropdownMenuClassName="dropdown-menu inverse"
+        dropdownMenuListClassName="dropdown-menu-list"
+        dropdownMenuListItemClassName="clickable"
+        initialID="more"
+        items={dropdownItems}
+        key="more"
+        onItemSelection={this.handleMoreDropdownSelection.bind(this, jobID)}
+        persistentID="more"
+        transition={true}
+        wrapperClassName="dropdown anchor-right" />
+    ];
+  }
+
+  getErrorScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center vertical-center inverse">
+        <RequestErrorMsg />
+      </div>
+    );
+  }
+
+  getLoadingScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center
+        vertical-center inverse">
+        <div className="row">
+          <div className="ball-scale">
+            <div />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  getNavigationTabs() {
+    return (
+      <ul className="tabs tall list-inline flush-bottom inverse">
+        {this.tabs_getUnroutedTabs()}
+      </ul>
+    );
+  }
+
+  getSubTitle(job) {
+    let longestRunningActiveRun = job.getActiveRuns()
+      .getLongestRunningActiveRun();
+    let longestRunningTask = longestRunningActiveRun.getTasks()
+      .getLongestRunningTask();
+    let dateCompleted = longestRunningTask.getDateCompleted();
+
+    let status = TaskStates[longestRunningTask.getStatus()];
+    let statusClasses = classNames('job-details-header-status', {
+      'text-success': status.stateTypes.indexOf('success') > -1
+        && status.stateTypes.indexOf('failure') === -1,
+      'text-danger': status.stateTypes.indexOf('failure') > -1
+    });
+
+    let timePrefix = null;
+    let shouldSuppressRelativeTime = dateCompleted == null;
+
+    if (shouldSuppressRelativeTime) {
+      timePrefix = 'for ';
+    }
+
+    return [
+      <span className={statusClasses} key="status-text">
+        {status.displayName}
+      </span>,
+      <span key="time-running">
+        ({timePrefix}{DateUtil.msToRelativeTime(
+          longestRunningActiveRun.getDateCreated(),
+          shouldSuppressRelativeTime
+        )})
+      </span>
+    ];
+  }
+
+  renderConfigurationTabView() {
+    return <JobConfiguration jobID={this.props.params.id} />;
```

Comment:
i'm cool with creating a separate component for this, but is there any reason why you don't just use the `DescriptionList` directly here? i don't think it's that much code to warrant a new component. but i'm cool with it like i said